source/languageHandler.py

```diff
@@ -213,85 +337,88 @@ def setLanguage(lang: str) -> None:
+		return False
 	except ValueError:
-		log.debugWarning(f"python locale {localeName} could not be retrieved with getlocale")
+		log.debugWarning(f"python locale {localeString} could not be retrieved with getlocale")
+		return False
 
-	if '-' in localeName:
-		# Python couldn't support the language-country locale, try language_country.
-		try:
-			localeName = localeName.replace('-', '_')
-			locale.setlocale(locale.LC_ALL, localeName)
-			locale.getlocale()
-			log.debug(f"set python locale to {localeName}")
-			return
-		except locale.Error:
-			log.debugWarning(f"python locale {localeName} could not be set")
-		except ValueError:
-			log.debugWarning(f"python locale {localeName} could not be retrieved with getlocale")
-
-	if '_' in localeName:
-		# Python couldn't support the language_country locale, just try language.
-		try:
-			localeName = localeName.split('_')[0]
-			locale.setlocale(locale.LC_ALL, localeName)
-			locale.getlocale()
-			log.debug(f"set python locale to {localeName}")
-			return
-		except locale.Error:
-			log.debugWarning(f"python locale {localeName} could not be set")
-		except ValueError:
-			log.debugWarning(f"python locale {localeName} could not be retrieved with getlocale")
 
+def setLocale(localeName: str) -> None:
+	'''
+	Set python's locale using a `localeName` such as "en", "ru_RU", or "es-ES".
+	Will fallback on `curLang` if it cannot be set and finally fallback to the system locale.
+	Passing NVDA locales straight to python `locale.setlocale` does now work since it tries to normalize the
+	parameter using `locale.normalize` which results in locales unknown to Windows (Python issue 37945).
+	For example executing: `locale.setlocale(locale.LC_ALL, "pl")`
+	results in locale being set to `('pl_PL', 'ISO8859-2')`
+	which is meaningless to Windows,
+	'''
+	originalLocaleName = localeName
+	localeString = ""
 	try:
-		locale.getlocale()
+		localeString = localeStringFromLocaleCode(localeName)
+		log.debug(f"Win32 locale string from locale code is {localeString}")
 	except ValueError:
-		# as the locale may have been changed to something that getlocale() couldn't retrieve
-		# reset to default locale
-		if originalLocaleName == curLang:
-			# reset to system locale default if we can't set the current lang's locale
-			locale.setlocale(locale.LC_ALL, "")
-			log.debugWarning(f"set python locale to system default")
+		log.debugWarning(f"Locale {localeName} not supported by Windows")
+	if localeString and _setPythonLocale(localeString):
+		return
+	else:
+		# The full form langName_country either cannot be retrieved from Windows
+		# or Python cannot be set to that locale.
+		# Try just with the language name.
+		if "_" in localeName:
+			localeName = localeName.split("_")[0]
+			try:
+				localeString = localeStringFromLocaleCode(localeName)
+				log.debug(f"Win32 locale string from locale code is {localeString}")
+			except ValueError:
+				log.debugWarning(f"Locale {localeName} not supported by Windows")

```

Comment:
This could also be simplified

```suggestion
	# The full form langName_country either cannot be retrieved from Windows
	# or Python cannot be set to that locale.
	# Try just with the language name.
	if "_" in localeName:
		localeName = localeName.split("_")[0]
		try:
			localeString = localeStringFromLocaleCode(localeName)
			log.debug(f"Win32 locale string from locale code is {localeString}")
		except ValueError:
			log.debugWarning(f"Locale {localeName} not supported by Windows")
```